### PR TITLE
Fixes #26119: All rule show up in directive compliance

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/26119

The directive compliance reports are using rules compliance reports, but currently an empty list of rule reports is just flattened further down as a rule with 0 compliance.
It should instead be deduced from the `Option`, obtained when getting directive reports

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala [hunk 1]
```diff
@@ -684,34 +690,43 @@ class ComplianceAPIService(
...
       } yield {
         // We will compute compliance for each rule for the current Directive
-        val rulesCompliance = for {
-          (ruleId, ruleReports) <- reportsByRule.toSeq
-
-          // We will now gather our report by component for the current Directive
-          reportsByComponents = (for {
-                                  ruleReport       <- ruleReports
-                                  nodeId            = ruleReport.nodeId
-                                  directiveReports <- ruleReport.directives.get(directive.id).toList
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala:694]: This is likely the culprit : the `option.toList` is hidden in the for-comprehension.
It exactly indicates if the rule has reports for the directive i.e. if the compliance report of this rule is needed or not.

**Comment by fanf** [comment]
can you please add a unit test ? It's clearly a edge case we want to have on chek

File: webapp/sources/rudder/rudder-rest/src/test/resources/api/api_compliance.yml [hunk 2]
```diff
@@ -2312,3 +2312,59 @@ response:
...
+            {
+              "id" : "br4",
+              "name" : "R4",
+              "compliance" : 100.0,
+              "policyMode" : "enforce",
+              "complianceDetails" : {
+                "successAlreadyOK" : 100.0
+              },
+              "components" : []
+            }
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-rest/src/test/resources/api/api_compliance.yml:2365]: in the previous version, it returned rules that did not event target this directive (only `br4` has this directive) : 
```
        +        {
        +          "id" : "br2",
        +          "name" : "R2",
        +          "compliance" : 0.0,
        +          "policyMode" : "enforce",
        +          "complianceDetails" : {},
        +          "components" : []
                 }
```

This test ensures that this no longer happens, only rules with the directive are listed in the directive compliance

**Review by fanf — APPROVED** [review]
seems to do as intended :+1:

**Comment by clarktsiory** [comment]
OK, squash merging this PR
